netlify/functions/member.ts

```diff
@@ -1,38 +1,99 @@
-import { Handler, HandlerEvent } from "@netlify/functions";
+import { Handler, HandlerContext, HandlerEvent } from "@netlify/functions";
+import { v2 as cloudinary } from "cloudinary";
+import { UploadApiResponse } from "cloudinary";
+import { parse } from "lambda-multipart-parser";
 
+import { loggedIn } from "./utils/auth";
 import { getFaunaClient } from "./utils/fauna";
 import { badRequest, ok } from "./utils/responses";
+import { Router } from "./utils/router";
 import { Document } from "./utils/types";
 
 import { Member } from "@/common/types/club";
 
 const { faunaClient, q } = getFaunaClient();
 
+const router = new Router("/api/member");
+
 /**
  *
  * GET /member/:email -> returns data for the member by email
  *
  */
+router.get("/:email", async (req) => {
+  if (!req.params.email) return badRequest("Missing email");
+  const email = req.params.email;
+
+  const faunaReq = await faunaClient.query<{ data: Document<Member>[] }>(
+    q.Map(
+      q.Paginate(q.Match(q.Index("members_by_email"), email)),
+      q.Lambda("X", q.Get(q.Var("X")))
+    )
+  );
+
+  const body = faunaReq.data[0].data;
+  return ok(JSON.stringify(body));
+});
+
+router.post("/avatar", loggedIn, async ({ event, context }) => {
+  const email = context.clientContext!.user!.email;
+  try {
+    // Parse the multipart/form-data request
+    const parsed = await parse(event);
+    if (!parsed.files.length) return badRequest("No file uploaded");
+
+    const avatarFile = parsed.files[0];
 
-const handler: Handler = async function (event: HandlerEvent) {
-  if (event.httpMethod === "GET") {
-    const pathArray = event.path.split("/");
-    const emailIndex = pathArray.indexOf("member") + 1;
-    const email = pathArray[emailIndex];
-
-    const req = await faunaClient.query<{ data: Document<Member>[] }>(
-      q.Map(
-        q.Paginate(q.Match(q.Index("members_by_email"), email)),
-        q.Lambda("X", q.Get(q.Var("X")))
-      )
+    const user = await faunaClient.query<Document<Member>>(
+      q.Get(q.Match(q.Index("members_by_email"), email))
     );
 
-    const responseBody = req.data[0].data;
+    // Upload to Cloudinary using the SDK
```

Comment:
Can we pull the cloudinary stuff outside of this file? Would be nice to have some clean interface for replacing the users photo that this api call makes, so that in the future if we replace cloudinary with another provider we can just replace it there, and the api call doesn’t need to know why

Reply:
That's a good point. I've created an `imageRepository` file with the functions to perform the upload/delete so we should be able to hot swap out that if needed in the future.

As for file limits, the file limit is 6MB and that comes from the payload limit on the lambda function. I've added a warning on the frontend if the user tries to upload a large file. I don't think much can be done on the backend however. The lambda just crashes if it receives a request that is too large.